src/geofabrics/bathymetry_estimation.py

```diff
@@ -0,0 +1,1576 @@
+            cross_sections_dict['mid_y'].append(y_array[i])
+
+            # record the length of the line segment
+            cross_sections_dict['length'].append(length)
+
+        cross_sections = geopandas.GeoDataFrame(cross_sections_dict,
+                                                crs=sampled_channel.crs)
+        return cross_sections
+
+    def _estimate_water_level_and_slope(self, cross_sections: geopandas.GeoDataFrame):
+        """ Estimate the water level and slope from the minimum z heights along
+        the sampled cross_sections after applying appropiate smoothing and
+        constraints to ensure it is monotonically increasing. Values are stored
+        in the cross_sections.
+
+        Parameters
+        ----------
+
+        cross_sections
+            The cross_sections with geometry defined as polylines.
+        """
+
+        # Min z values as the water surface. Ensure no NaN
+        cross_sections['min_z_centre'] = cross_sections['min_z_centre'].interpolate('index', limit_direction='both')
+        # Unimodal enforaces monotonically increasing
+        cross_sections['min_z_centre_unimodal'] = self._unimodal_smoothing(cross_sections['min_z_centre'])
+
+        # Set the water z value to use for width thresholding
+        cross_sections['min_z_water'] = cross_sections['min_z_centre_unimodal']
+
+        # Slope from the water surface - interpolate to fill any Nan
+        cross_sections['slope'] = cross_sections['min_z_water'].diff() / self.cross_section_spacing
+        cross_sections['slope'] = cross_sections['slope'].interpolate('index', limit_direction='both')
+
+        # Slopes for a range of smoothings
+        for smoothing_distance in [500, 1000, 2000, 3000]:
+            # ensure odd number of samples so array length preserved
+            smoothing_samples = int(numpy.ceil(smoothing_distance / self.cross_section_spacing))
+            smoothing_samples = int(smoothing_samples / 2) * 2 + 1
+            label = f'{smoothing_distance/1000}km'
+
+            # Smoothed min_z_centre_unimodal
+            cross_sections[f'min_z_centre_unimodal_mean_{label}'] \
+                = self._rolling_mean_with_padding(cross_sections['min_z_centre_unimodal'],
+                                                  smoothing_samples)
+
+            # Smoothed slope
+            cross_sections[f'slope_mean_{label}'] = self._rolling_mean_with_padding(cross_sections['slope'],
+                                                                                    smoothing_samples)
+
+    def _smooth_widths_and_thresholds(self, cross_sections: geopandas.GeoDataFrame):
+        """ Record the valid and reolling mean of the calculated thresholds and widths
+
+        Parameters
+        ----------
+
+        cross_sections
+            The elevations and calculated widths and thresholds for each sampled cross section
+        """
+
+        invalid_mask = cross_sections['valid'] == False
```

Comment:
```suggestion
        invalid_mask = not cross_sections['valid']
```
Would this be more clear to you or less? Or even changing it to `valid_mask` and keeping it like that

Reply:
PEP E712 comparison to False should be 'if cond is False:' or 'if not cond:'

Reply:
If you change anything here, there's a couple other instances of the same in this file

Reply:
Thanks @LukeParky your suggestion of np.logical_not was great. I've changed all my comparison to false to this.